### DISCUSSION TITLE
PutObject - file not found error

### DISCUSSION THREAD

**Discussion by nebnes** [minio/minio#14954] [Q&A]
Hello, 
I am strangle with a "file not found" error on a PutObject request which seems strange. I don't know if my problem comes from Minio (which I use for local development), from Rusoto (my s3 rust client) or from my side.

Do you have any clue about this type of error?

Here is the Minio log:
```
API: PutObject
Time: Fri May 20 2022 15:19:41 GMT+0200 (Central European Summer Time)
DeploymentID: d85dab42-d42a-42f1-af1e-532bc215e3c0
RequestID: 16F0D94B6D4D1E40
RemoteHost: 172.18.0.58
UserAgent: rusoto/0.47.0 rust/1.60.0 linux
Error: file not found (cmd.StorageErr)

Backtrace: 
0: cmd/fs-v1-helpers.go:360:cmd.fsRenameFile()
1: cmd/fs-v1.go:1107:cmd.(*FSObjects).putObject()
2: cmd/fs-v1.go:1012:cmd.(*FSObjects).PutObject()
3: cmd/object-handlers.go:1826:cmd.objectAPIHandlers.PutObjectHandler()
4: net/http/server.go:2047:http.HandlerFunc.ServeHTTP()
```

Rusoto return by error :
```
Request ID: None Body: 
<?xml version="1.0"?>
<Error>
  <Code>NoSuchKey</Code>
  <Message>The specified key does not exist.</Message>
  <Key>folder/a25eeccd-dff1-49d2-87f1-efc2f8470ed3.jpg</Key>
  <BucketName>test</BucketName>
  <Resource>/test/folder/a25eeccd-dff1-49d2-87f1-efc2f8470ed3.jpg</Resource>
  <RequestId>16F0D94B6D4D1E40</RequestId>
  <HostId>d85dab42-d42a-42f1-af1e-532bc215e3c0</HostId>
</Error>
```

Thanks in advance

**Comment by harshavardhana** [comment]
which version of MinIO is this? and what is your backend disk here?

**Reply by nebnes** [reply]
I run MinIO with docker image : `minio/minio:RELEASE.2022-05-08T23-50-31Z.fips`
and the disk of the host is a SSD.

**Reply by harshavardhana** [reply]
what is the filesystem? @nebnes

**Reply by nebnes** [reply]
The volume is under : `ext4`

**Reply by harshavardhana** [reply]
This looks like a filesystem corruption, we can't do much here..

We wrote something in synchronous mode, however it doesn't exist on the namespace when a rename is attempt to move it actual entry on the namespace.

**Reply by nebnes** [reply]
@harshavardhana I think I found the source of my problem : 
I realized that minio was deleting the empty folders. Apparently during a PutObject request almost at the same time minio was deleting the target folder. 

My solution was to  uploaded empty files in the folders to prevent minio 
from deleting the folders when they are empty. I don't have the error anymore. 

Is it a specific behavior of minio (vs aws s3) ?

**Comment by Vidyasagar9390** [comment]
Hey hai @nebnes 

how you handle this error, could be specific 
i also got the same error 
I want fix
